### PR TITLE
feat: support parameterized query sdk

### LINKED ISSUE #168: SqlRouter support Execute Batch SQL with ParameterRow/Request Row
**Is your feature request related to a problem? Please describe.**

We plan to support execute SQL with given Parameters (RequestRow)  under `batch-mode` .
**Describe the solution you'd like**

```c++
std::shared_ptr<::hybridse::sdk::ResultSet> SQLClusterRouter::ExecuteSQL(
    const std::string& db, const std::string& sql, std::shared_ptr<openmldb::sdk::SQLRequestRow> parameter,
    ::hybridse::sdk::Status* status) {
    // TODO
}


bool TabletClient::Query(const std::string& db, const std::string& sql,
                         const std::vector<openmldb::type::DataType>& parameter_types,
                         const std::string& parameter_row,
                         brpc::Controller* cntl, ::openmldb::api::QueryResponse* response, const bool is_debug) {
// TODO
}
```
**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by jingchen2222**
* **Please check if the PR fulfills these requirements**

- The commit message follows our guidelines
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

feature and bug fix.

* **What is the current behavior?** (You can also link to an open issue here)
#168 

* **What is the new behavior (if this is a feature change)?**

  - CICD: allow openmldb to use hybridse build from local https://github.com/4paradigm/OpenMLDB/pull/248/commits/03cdb04366e75963bf89b3e3f72c7fcdd8a10561
  - BUILD: fix: mini cluster core on mac. https://github.com/4paradigm/OpenMLDB/pull/248/commits/d7c5e6c91294a9bbe720159ffefbf4f9b931a11d
  - Hybridse: fix engine cache trigger by parameterized query. https://github.com/4paradigm/OpenMLDB/pull/248/commits/2d2d6d21631503df274bde72e923781deb32a693
  - Hybridse: refactor Session interface, only BatchRunSession can support parameterized run method.https://github.com/4paradigm/OpenMLDB/pull/248/commits/29e268076fd7b6581456bcac0ff6fff5ca780dc8
  - Hybridse: remove `session.Run()`  interface.  https://github.com/4paradigm/OpenMLDB/pull/248/commits/76542f70073b0ebb48b046a1592b95081c53a638
  - C++ SDK: insert sdk bug fix for parameterized insert. https://github.com/4paradigm/OpenMLDB/pull/248/commits/d59f5dc1eb7b5a411212de33e0318323c0262f98
  - C++ SDK: support parameterized query. https://github.com/4paradigm/OpenMLDB/pull/248/commits/05989329ad953f060ba1ecd79607656906a536aa
  - TEST: fix:UninstantiatedParameterizedTestSuite error. https://github.com/4paradigm/OpenMLDB/pull/248/commits/90dc3aeb9f90a0d76eaed419ba010fe342c17cc0

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
  - C++ SDK: Method change. `ExecuteSQL(db, sql,request)` change to `ExecuteRequestSQL(db, sql,request)`
  - C++ SDK: Add Method. `ExecuteSQL(db, sql, parameter_schema, parameter)`

### Demo
```c++
// ... ignore
    hybridse::codec::Schema parameter_schema;
    {
        auto column = parameter_schema.Add();
        column->set_type(::hybridse::type::kVarchar);
    }
    {
        auto column = parameter_schema.Add();
        column->set_type(::hybridse::type::kInt64);
    }
    const std::shared_ptr<::hybridse::sdk::Schema> schema_impl = std::make_shared<hybridse::sdk::SchemaImpl>(schema);
    std::string where_exist = "select * from trans where merch_id = ? and txn_time < ?;";
    // parameterized query
    auto parameter_row = std::make_shared<SQLRequestRow>(schema_impl, std::set<std::string>());
    {
        ASSERT_EQ(2, parameter_row->GetSchema()->GetColumnCnt());
        ASSERT_TRUE(parameter_row->Init(4));
        ASSERT_TRUE(parameter_row->AppendString("mc_0"));
        ASSERT_TRUE(parameter_row->AppendInt64(1594800959830));
        ASSERT_TRUE(parameter_row->Build());

        auto rs = router->ExecuteSQL(db, where_exist, parameter_row, &status);

        if (!rs) {
            FAIL() << "fail to execute sql";
        }
        ASSERT_EQ(rs->Size(), 3);
    }
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Changelog
+
+## [Unreleased]
+
+### Feature
+- Support parameterized query under BatchMode [#262](https://github.com/4paradigm/OpenMLDB/issues/262), [#168](https://github.com/4paradigm/OpenMLDB/issues/168)
+
+### SQL Syntax
+- `nvl` & `nvl2`: [#238](https://github.com/4paradigm/OpenMLDB/issues/238)
```
Comment (aceforeverd) [inline CHANGELOG.md:9]: missing `bitwise`

File: CPPLINT.cfg [hunk 2]
```diff
@@ -1 +1,2 @@
 linelength=120
+filter=-build/header_guard
```
Comment (aceforeverd) [inline CPPLINT.cfg]: should we tweak header gurads in hybridse?

File: hybridse/include/base/fe_status.h [hunk 3]
```diff
@@ -27,10 +27,14 @@ namespace base {
 template <typename STREAM, typename... Args>
 static inline std::initializer_list<int> __output_literal_args(STREAM& stream,  // NOLINT
                                                                Args... args) {  // NOLINT
+#ifdef __APPLE__
```
Comment (aceforeverd) [inline hybridse/include/base/fe_status.h:30]: mac need ?

File: .github/workflows/cicd.yaml [hunk 4]
```diff
@@ -62,18 +63,17 @@ jobs:
...
         if: always()
         run: |
-          source /etc/profile.d/enable-rh.sh
           git clean -dfx
 
   sql_sdk_test:
     runs-on: ubuntu-latest
     container:
-      image: ghcr.io/4paradigm/hybridsql:0.3.0
+      image: ghcr.io/4paradigm/hybridsql:0.4.0
```
Comment (aceforeverd) [inline .github/workflows/cicd.yaml]: well, we have conflicts #266

File: src/sdk/mini_cluster.h [hunk 5]
```diff
@@ -55,7 +55,9 @@ namespace openmldb {
 namespace sdk {
 
 constexpr int MAX_TABLET_NUM = 3;
+#ifdef __linux__
```
Comment (aceforeverd) [inline src/sdk/mini_cluster.h:58]: is mac broken ?

File: hybridse/include/base/fe_status.h (hunk 3 above)
Comment (jingchen2222) [reply hybridse/include/base/fe_status.h:30]: Yes. Ignore warnings in mac

File: .github/workflows/cicd.yaml (hunk 4 above)
Comment (jingchen2222) [reply .github/workflows/cicd.yaml]: Yep. We will sync the update after merging #266

File: src/sdk/mini_cluster.h (hunk 5 above)
Comment (jingchen2222) [reply src/sdk/mini_cluster.h:58]: Yes.  It fix bprc server core on mac . Check #247 for more details.

File: .github/workflows/cicd.yaml [hunk 6]
```diff
@@ -11,6 +11,7 @@ on:
 
 env:
   GIT_SUBMODULE_STRATEGY: recursive
+  HYBRIDSE_SOURCE: download
```
Comment (aceforeverd) [inline .github/workflows/cicd.yaml:14]: download ?

**Review by aceforeverd — APPROVED** [review]

**Review by imotai — APPROVED** [review]

**Pull request merged**